.github/workflows/test.yml

```diff
@@ -132,6 +132,12 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
```

Comment:
Here and lines 255-259 it would make sense to put it directly before the `docker-compose` call to make it easier to correlate and see hte connection between the two.